content/en/apps/guides/performance/replication.md

```diff
@@ -37,10 +39,108 @@ For example:
 }
 ```
 
+##### Code sample 3.10 and later:
+```json
+{
+  "replication_depth": [
+    { "role": "district_manager", "depth": 1, "report_depth": 1 },
+    { "role": "national_manager", "depth": 2 }
+  ]
+}
+```
+
+##### Contact Depth
+
+Contact depth is calculated relative to the user's home place, having the user's facility at depth = 0. Places and contacts that are parented by the facility are on depth = 1, and so on. 
+
+For example, considering the following contact hierarchy:
+```
+district > health_center > clinic > area > family > patient
+``` 
+a `supervisor` at `health_center` level, and the following configuration:
+```json
+{ "replication_depth": [{ "role": "supervisor_role", "depth": 2 }]}
+``` 
+
+The `supervisor` would download the following contacts and reports:
+- (level 0) `health_center` + all reports about `health_center`
+- (level 1) contacts whose parent is `health_center` +  all reports about these contacts
+- (level 1) `clinic` + all reports about `clinic`
+- (level 2) contacts whose parent is `clinic` + all reports about these contacts
+- (level 2) `family` + all reports about `family`
+
+With the following configuration:
+```json
+{ "replication_depth": [{ "role": "supervisor_role", "depth": 1 }]}
+``` 
+the `supervisor`'s docs change to:
+- (level 0) `health_center` + all reports about `health_center`
+- (level 1) contacts whose parent is `health_center` + all reports about these contacts
+- (level 1) `clinic` + all reports about `clinic` 
+
+##### Report Depth
+
+As of 3.10, `report_depth` works similarly to `depth`, but it's applied to reports (documents with `"type": "data_record"`) alone, excepting reports that are submitted by the authenticated user. 
+It only works in conjunction with `depth`.   
+
+Following the examples above:
+```json
+{ "replication_depth": [{ "role": "supervisor_role", "depth": 2, "report_depth": 1 }]}
+```
+
+The `supervisor` would download the following contacts and reports:
+- (level 0) `health_center` + all reports about `health_center`
+- (level 1) contacts whose parent is `health_center` + all reports about these contacts
+- (level 1) `clinic` + all reports about `clinic`
+- (level 2) contacts whose parent is `clinic` + reports about `clinic` submitted by `supervisor`
+- (level 2) `family` +  reports about `family` submitted by the `supervisor`
+
+Similarly, with:
+```json
+{ "replication_depth": [{ "role": "supervisor_role", "depth": 2, "report_depth": 0 }]}
+``` 
+the `supervisor`'s docs change to:
+- (level 0) `health_center` + all reports about `health_center`
+- (level 1) contacts whose parent is `health_center` + reports about these contacts submitted by `supervisor`
+- (level 1) `clinic` + reports about `clinic` submitted by the `supervisor`
+- (level 2) contacts whose parent is `clinic` + reports about these contacts submitted by `supervisor`
+- (level 2) `family` + reports about `family` submitted by the `supervisor`
```

Comment:
These examples are great, but they're hard to visualise in text form. Do you think it would be possible to have a table with ticks and crosses to make it easier to compare depths? eg:

Docs | Config A | Config B | Config C | Config D
--|--|--|--|--
`health_center` | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark:
Reports about `health_center` | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :x:

Reply:
Done. The report replication depth table(s) is huge :( (I split it in 2 halves, so you'd have a visible header within the screen).